editors/code/src/commands/inlay_hints.ts

```diff
@@ -83,17 +83,53 @@ export class HintsUpdater {
     ): Promise<void> {
         const newHints = await this.queryHints(editor.document.uri.toString());
         if (newHints !== null) {
-            const newDecorations = newHints.map(hint => ({
-                range: hint.range,
-                renderOptions: { after: { contentText: `: ${hint.label}` } }
-            }));
+            const newDecorations = newHints.map(hint => {
+                const [label, range] = this.truncateHint(
+                    hint.label,
+                    hint.range
+                );
+                return {
+                    range,
+                    renderOptions: {
+                        after: {
+                            contentText: `: ${label}`
+                        }
+                    }
+                };
+            });
             return editor.setDecorations(
                 typeHintDecorationType,
                 newDecorations
             );
         }
     }
 
+    private truncateHint(
+        label: string,
+        range: vscode.Range
+    ): [string, vscode.Range] {
+        if (!Server.config.maxInlayHintLength) {
+            return [label, range];
+        }
+
+        let newLabel = label.substring(0, Server.config.maxInlayHintLength);
+        if (label.length > Server.config.maxInlayHintLength) {
+            newLabel += '…';
+        }
+
+        range = new vscode.Range(
```

Comment:
Why do we need to modify the `range` here? It's the range of source code for which we produce a hint, and is unrelated tot the hint itself I think.

Reply:
Oops, you're right. Fixed!